src/store/RoutingStore.js

```diff
@@ -0,0 +1,26 @@
+import { action } from 'mobx';
+
+const allDialogs = [];
+
+export default class RoutingStore {
+    @action.bound handleRouting() {
+        let timer;
+        window.addEventListener('hashchange', () => {
+            clearTimeout(timer);
+            timer = setTimeout(() => {
+                const hash = window.history.state.urlPath.replace('#', '');
+                if (hash === '') {
+                    this.closeAll();
+                }
+            }, 200);
+        }, false);
```

Comment:
Why do we need the setTimeout? the event is async itself.

Remove the timeout.